### PR TITLE
Allow fail for tags

### PR DESCRIPTION
**Pull request by maximlopin**
- training a predictor can fail for `TAGS` data type

### CODE REVIEW

**Comment by ZoranPandovski** [comment]
**CLA Assistant Lite** All Contributors have signed the CLA.

**Comment by ZoranPandovski** [comment]
**CLA Assistant Lite** All Contributors have signed the CLA.

**Comment by ZoranPandovski** [comment]
**CLA Assistant Lite** All Contributors have signed the CLA.

**Comment by ZoranPandovski** [comment]
**CLA Assistant Lite** All Contributors have signed the CLA.

File: mindsdb_native/libs/phases/model_analyzer/model_analyzer.py [hunk 1]
```diff
@@ -43,8 +46,9 @@ def run(self):
             
             if data_type == DATA_TYPES.CATEGORICAL:
                 if data_subtype == DATA_SUBTYPES.TAGS:
-                    # need to implement stats_v2[col]['guess_probability'] for tags
-                    pass
+                    encoder = self.transaction.model_backend.predictor._mixer.encoders[col]
+                    if accuracy_score(encoder.encode(reals), encoder.encode(preds)) < self.transaction.lmd['stats_v2'][col]['guess_probability']:
+                        fails = True
```
Comment (paxcema) [inline mindsdb_native/libs/phases/model_analyzer/model_analyzer.py:48]: I think `acc=guess_probability` should also be considered a failure

**Comment by ZoranPandovski** [comment]
**CLA Assistant Lite** All Contributors have signed the CLA.

**Comment by ZoranPandovski** [comment]
**CLA Assistant Lite** All Contributors have signed the CLA.

File: mindsdb_native/libs/phases/model_analyzer/model_analyzer.py [hunk 2]
```diff
@@ -1,7 +1,10 @@
 from mindsdb_native.libs.helpers.general_helpers import pickle_obj, disable_console_output
 from mindsdb_native.libs.constants.mindsdb import *
 from mindsdb_native.libs.phases.base_module import BaseModule
-from mindsdb_native.libs.helpers.general_helpers import evaluate_accuracy
+from mindsdb_native.libs.helpers.general_helpers import (
+    evaluate_accuracy,
+    evaluate_multilabel_accuracy
```
Comment (paxcema) [inline mindsdb_native/libs/phases/model_analyzer/model_analyzer.py]: Seems like `evaluate_multilabel_accuracy` is not being used?

**Review by paxcema — APPROVED** [review]

**Comment by ZoranPandovski** [comment]
**CLA Assistant Lite** All Contributors have signed the CLA.

**Pull request merged**